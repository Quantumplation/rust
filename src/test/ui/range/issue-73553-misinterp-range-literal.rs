type Range = std::ops::Range<usize>;

fn demo(r: &Range) {
    println!("{:?}", r);
}

fn tell(x: usize) -> usize {
    x
}

fn main() {
    demo(tell(1)..tell(10));
    //~^ ERROR arguments to this function are incorrect
    demo(1..10);
    //~^ ERROR arguments to this function are incorrect
}
