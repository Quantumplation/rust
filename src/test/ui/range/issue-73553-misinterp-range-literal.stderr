error[E0308]: arguments to this function are incorrect
  --> $DIR/issue-73553-misinterp-range-literal.rs:12:5
   |
LL |     demo(tell(1)..tell(10));
   |     ^^^^^-----------------^
   |          |
   |          expected `&std::ops::Range<usize>`, found `std::ops::Range<_>`
   |
note: function defined here
  --> $DIR/issue-73553-misinterp-range-literal.rs:3:4
   |
LL | fn demo(r: &Range) {
   |    ^^^^ ---------
help: provide an argument of the correct type
   |
LL |     demo({&std::ops::Range<usize>});
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/issue-73553-misinterp-range-literal.rs:14:5
   |
LL |     demo(1..10);
   |     ^^^^^-----^
   |          |
   |          expected `&std::ops::Range<usize>`, found `std::ops::Range<_>`
   |
note: function defined here
  --> $DIR/issue-73553-misinterp-range-literal.rs:3:4
   |
LL | fn demo(r: &Range) {
   |    ^^^^ ---------
help: provide an argument of the correct type
   |
LL |     demo({&std::ops::Range<usize>});
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
